src/migrateUIFolder.test.ts

```diff
@@ -1,16 +1,73 @@
+import _map from "lodash/map";
+import _intersection from "lodash/intersection";
+import _some from "lodash/some";
 import { migrateUIFolder } from "./migrateUIFolder";
 import { smallLegacyUIFolder } from "./__test_resources__/smallLegacyUIFolder";
 import { legacyUIFolder } from "./__test_resources__/legacyUIFolder";
 import { servers } from "./__test_resources__/servers";
+import { ContentRecord } from "@activeviam/activeui-sdk";
 
 describe("migrateUIFolder", () => {
-  it("returns a valid ActiveUI5 /ui folder on a small input", async () => {
+  it("returns a valid ActiveUI5 /ui folder on a small input", () => {
     const migratedUIFolder = migrateUIFolder(smallLegacyUIFolder, servers);
     expect(migratedUIFolder).toMatchSnapshot();
   });
 
-  it("returns a valid ActiveUI5 /ui folder on a real life input", async () => {
+  it("returns a valid ActiveUI5 /ui folder on a real life input", () => {
     const migratedUIFolder = migrateUIFolder(legacyUIFolder, servers);
     expect(migratedUIFolder).toMatchSnapshot();
   });
+
+  it("removes the specified widget plugins from the widget bookmarks themselves, and from the content of the dashboard bookmarks", () => {
+    const keysOfWidgetPluginsToRemove = ["filters"];
+    const migratedUIFolder = migrateUIFolder(
+      legacyUIFolder,
+      servers,
+      keysOfWidgetPluginsToRemove
+    );
+
+    // In the Activeui 4 folder, the file with id `0xb` represents a saved Page Filters widget.
+    // It is removed from the ActiveUI 5 UI folder.
+    const savedWidgets = migratedUIFolder.children!.widgets;
+    const hasRecord = (
+      contentRecord: ContentRecord,
+      recordId: string
+    ): boolean =>
+      _some(
+        contentRecord.children,
+        (child, childId) => childId === recordId || hasRecord(child, recordId)
+      );
+    expect(hasRecord(savedWidgets, "0xb")).toBe(false);
```

Comment:
It would also be nice here to check that the legacy UI folder has a widget with this id.